tower-http/src/cors.rs

```diff
@@ -0,0 +1,809 @@
+        };
+        self
+    }
+
+    /// Set the value of the [`Access-Control-Allow-Origin`][mdn] header.
+    ///
+    /// ```
+    /// use tower_http::cors::{CorsLayer, Origin};
+    ///
+    /// let layer = CorsLayer::new().allow_origin(Origin::exact(
+    ///     "http://example.com".parse().unwrap(),
+    /// ));
+    /// ```
+    ///
+    /// Multiple origins can be allowed with
+    ///
+    /// ```
+    /// use tower_http::cors::{CorsLayer, Origin};
+    ///
+    /// let origins = vec![
+    ///     "http://example.com".parse().unwrap(),
+    ///     "http://api.example.com".parse().unwrap(),
+    /// ];
+    ///
+    /// let layer = CorsLayer::new().allow_origin(Origin::list(origins));
+    /// ```
+    ///
+    /// All origins can be allowed with
+    ///
+    /// ```
+    /// use tower_http::cors::{CorsLayer, Any};
+    ///
+    /// let layer = CorsLayer::new().allow_origin(Any);
+    /// ```
+    ///
+    /// You can also use a closure
+    ///
+    /// ```
+    /// use tower_http::cors::{CorsLayer, Origin};
+    /// use http::{HeaderValue, request::Parts};
+    ///
+    /// let layer = CorsLayer::new().allow_origin(
+    ///     Origin::predicate(|origin: &HeaderValue, _request_head: &Parts| {
+    ///         origin.as_bytes().ends_with(b".rust-lang.org")
+    ///     })
+    /// );
+    /// ```
+    ///
+    /// Note that multiple calls to this method will override any previous
+    /// calls.
+    ///
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin
+    pub fn allow_origin<T>(mut self, origin: T) -> Self
+    where
+        T: Into<AnyOr<Origin>>,
+    {
+        self.allow_origin = Some(origin.into());
+        self
+    }
+
+    /// Set the value of the [`Access-Control-Expose-Headers`][mdn] header.
+    ///
+    /// ```
+    /// use tower_http::cors::CorsLayer;
+    /// use http::header::CONTENT_ENCODING;
+    ///
+    /// let layer = CorsLayer::new().expose_headers(vec![CONTENT_ENCODING]);
+    /// ```
+    ///
+    /// All headers can be allowed with
+    ///
+    /// ```
+    /// use tower_http::cors::{CorsLayer, Any};
+    ///
+    /// let layer = CorsLayer::new().expose_headers(Any);
+    /// ```
+    ///
+    /// Note that multiple calls to this method will override any previous
+    /// calls.
+    ///
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers
+    pub fn expose_headers<I>(mut self, headers: I) -> Self
+    where
+        I: Into<AnyOr<Vec<HeaderName>>>,
+    {
+        self.expose_headers = Some(match headers.into().0 {
+            AnyOrInner::Any => WILDCARD.parse().unwrap(),
+            AnyOrInner::Value(headers) => separated_by_commas(headers),
+        });
+        self
+    }
+}
+
+/// Represents a wildcard value (`*`) used with some CORS headers such as
+/// [`CorsLayer::allow_methods`].
+#[derive(Debug, Clone, Copy)]
+pub struct Any;
```

Comment:
```suggestion
#[derive(Debug, Default, Clone, Copy)]
pub struct Any(());
```

Reply:
I went with a `fn any() -> Any { Any }` instead which is a little easier to type 👉 2f9b5985f49f9